File: app/src/main/java/com/imangazalievm/bubbble/data/repository/ImagesRepository.java
package com.imangazalievm.bubbble.data.repository;

import com.imangazalievm.bubbble.data.filesystem.UrlImageSaver;
import com.imangazalievm.bubbble.domain.repository.IImagesRepository;

import javax.inject.Inject;

import io.reactivex.Completable;

public class ImagesRepository implements IImagesRepository {

    private UrlImageSaver urlImageSaver;

    @Inject
    public ImagesRepository(UrlImageSaver urlImageSaver) {
        this.urlImageSaver = urlImageSaver;
    }

    public Completable saveImage(String shotImageUrl) {
        return Completable.create(e -> {
            urlImageSaver.saveImage(shotImageUrl);
            e.onComplete();
        });
    }

}

File: core/data/src/main/java/com/bubbble/data/images/ImagesRepository.kt
package com.bubbble.data.images

import com.bubbble.data.global.filesystem.UrlImageSaver
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class ImagesRepository @Inject constructor(
    private val urlImageSaver: UrlImageSaver
) {

    suspend fun saveImage(shotImageUrl: String) = withContext(Dispatchers.IO) {
        urlImageSaver.saveImage(shotImageUrl)
    }

}